graphql_compiler/query_formatting/common.py

```diff
@@ -34,47 +34,43 @@ def validate_argument_type(expected_type, value, name=None, is_inner_value=False
         value: object that can be interpreted as being of that type
         name: optional string. If it is a string, it is the name of the argument and will be used to
               provide more descriptive error messages.
-        is_inner_value: bool, whether this value corresponds to an element in a list argument.
     """
-    # TODO(pmantica1): make name parameter required in major version bump.
-    if name is not None:
-        if is_inner_value:
-            error_message_prefix = u'Invalid type for element in list argument {}.'.format(name)
-        else:
+    def _get_invalid_type_error_message(expected_python_type_name):
+        """Return a string stating that the argument type is invalid."""
+        # TODO(pmantica1): Make name param required in valida_argument_type in major version bump.
+        if name is not None:
             error_message_prefix = u'Invalid type for argument {}.'.format(name)
-    else:
-        if is_inner_value:
-            error_message_prefix = u'Found element in list argument with invalid type.'
         else:
             error_message_prefix = u'Found element with invalid type.'
-    error_message_suffix = u'Got value {} of type {}.'.format(value, type(value).__name__)
-    invalid_type_error_message = error_message_prefix + ' Expected {}. ' + error_message_suffix
+        error_message_suffix = u'Got value {} of type {}.'.format(value, type(value).__name__)
+        error_message_middle = ' Expected {}. '.format(expected_python_type_name)
+        return error_message_prefix + error_message_middle + error_message_suffix
 
     stripped_type = strip_non_null_from_type(expected_type)
     if GraphQLString.is_same_type(stripped_type):
         if not isinstance(value, six.string_types):
-            raise GraphQLInvalidArgumentError(invalid_type_error_message.format('string'))
+            raise GraphQLInvalidArgumentError(_get_invalid_type_error_message('string'))
     elif GraphQLID.is_same_type(stripped_type):
         # IDs can be strings or numbers, but the GraphQL library coerces them to strings.
         # We will follow suit and treat them as strings.
         if not isinstance(value, six.string_types):
-            raise GraphQLInvalidArgumentError(invalid_type_error_message.format('string'))
+            raise GraphQLInvalidArgumentError(_get_invalid_type_error_message('string'))
```

Comment:
Seems like we always raise an error with the output of that function – why not make the function raise the error as well?